src/core/Akka.Streams/Queue.cs

```diff
@@ -59,6 +59,13 @@ public interface ISourceQueueWithComplete<in T> : ISourceQueue<T>
         /// </summary>
         /// <param name="ex">TBD</param>
         void Fail(Exception ex);
+
+        /// <summary>
+        /// Method returns <see cref="Task"/> that will be completed if the stream completes,
+        /// or will be failed when the stage faces an internal failure or the the <see cref="Fail"/> method is invoked.
+        /// </summary>
+        /// <returns>Task</returns>
+        new Task WatchCompletionAsync();
```

Comment:
`ISourceQueueWithComplete.WatchCompletionAsync` has different semantic that simple `ISourceQueue.WatchCompletionAsync`. What is the best way to highlight it?

Reply:
Seems like a @Silv3rcircl3 question